My project/Assets/Scripts/CreatureLogic/Actions/Breeding/Breed.cs

```diff
@@ -0,0 +1,122 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Breed : IAction
+{
+    //class variables
+    BaseCreature target;
+    CreatureData data;
+    RangeScanner scanner;
+    Rigidbody2D rb;
+    float breedDuration;
+    float breedTimeEnd;
+    float nextBreedTime;
+    float breedCooldown;
+    bool canBreed;
+
+    public void PrintStatus()
+    {
+        
+    }
+
+    public Breed()
+    {
+        //breed duration lasts a quarter traits may further change these
+        breedDuration = TimeManager.Instance.secondsPerDay / 4;
+        breedCooldown = TimeManager.Instance.secondsPerDay;
+        breedTimeEnd = 0;
+        nextBreedTime = 0;
+    }
+
+    //creature in range that has met the conditions of breedable
+    //creature in range is in breed action, and creature is willing to breed
+    //creature has already bred, then it is unable to breed according to its cooldown
+    public bool StartCondition()
+    {
+        if (Time.time < nextBreedTime)
+        {
+            //Debug.Log("Breed On Coowndown");
+            return false;
+        }
+
+        foreach(BaseCreature creature in scanner.GetCreatures())
+        {
+            string currentAction = creature.currentActionNode.action.ToString();
+            if (creature.data.CurrentEnergy >= creature.data.Energy / 2
+                || currentAction.Equals("LookForMate")
+                || currentAction.Equals("Breed")) 
+            {
+                target = creature;
+                //Debug.Log("Breeder " + data.ID + " |Breed Victim" + target.data.ID);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    //Set time limit of breed
+    public void OnEnter()
+    {
+        breedTimeEnd = Time.time + breedDuration;
+        canBreed = false;
+        rb.velocity = new Vector2(target.transform.position.x - rb.position.x, target.transform.position.y - rb.position.y).normalized * data.Speed;
+    }
+
+    //if check if the target is going to breed
+    //else do nothing
+    public void Run()
+    {
+        if (target.currentActionNode.action.ToString().Equals("Breed"))
+        {
+            //Debug.Log("Can Breed");
+            canBreed = true;
+        }
```

Comment:
my thought is that this should be closer to the evaluation on line 85. otherwise we might check that canBreed=true which falls out of sync with the actual target's breed state when we're checking proximity.

If it makes sense, i'd be fine with leaving Run() empty and putting all of this logic into EndCondition